### PR TITLE
Allow use of STREAM in snowflake CHANGES clause

### PR DESCRIPTION
**Pull request by alanmcruickshank**
Does what it says on the tin. Looks like we missed one keyword during the initial implementation https://github.com/sqlfluff/sqlfluff/pull/2764.

This resolves an issue raised on slack.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_snowflake.py [hunk 1]
```diff
@@ -1523,14 +1523,16 @@ class ChangesClauseSegment(BaseSegment):
             "INFORMATION",
             Ref("ParameterAssignerSegment"),
             OneOf("DEFAULT", "APPEND_ONLY"),
+            parse_mode=ParseMode.GREEDY,
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_snowflake.py:1526]: Why are these needed now out of interest?

**Review by tunetheweb — APPROVED** [review]

File: src/sqlfluff/dialects/dialect_snowflake.py (hunk 1 above)
Comment (alanmcruickshank) [reply src/sqlfluff/dialects/dialect_snowflake.py:1526]: Not needed at all, then just mean slightly more specific parsing errors if snowflake change the syntax again. I'm guessing some of those keywords might change and if they do - only the bracketed section that has changed will fail and we'll get a faster report on the issue.
